rfcs/convergence/no-css-shorthands-in-make-styles.md

```diff
@@ -0,0 +1,218 @@
+- ❌ it does not work when `--foo` is compound value, for example `4px 8px`
+
+[🔗 CodeSandbox demo](https://codesandbox.io/s/inline-style-expand-shorthand-css-variables-n9mh3)
+
+## Requirements
+
+### Stay with deterministic Atomic CSS
+
+It's important for scaling our library to stay with deterministic Atomic classes. For example, we can extract common styles to a stylesheet that will be used across all products.
+
+### Unsupported styles should fail at build, not runtime
+
+There were multiple options covered in the original issue (see [microsoft/fluentui#19402](https://github.com/microsoft/fluentui/issues/19402)), but during offline discussion we agreed to go with build time checks instead runtime and rely on TypeScript typings.
+
+The motivation is simple: no one sees console noise, in big applications it usually 100+ development warnings only on initial load.
+
+### Keep CSS properties
+
+- Keep interop with any CSS var on any CSS property
+- Keep all style properties to spec with CSS properties
+
+This means that custom syntax is a **discarded solution**, for example:
+
+```ts
+// ⚠ discarded solution, used only for an example
+makeStyles({
+  rootA: {
+    // 1️⃣ no custom properties
+    paddingX: 'var(--foo)',
+  },
+  rootB: {
+    // 2️⃣ no custom values
+    padding: { left: 'var(--foo)', right: 'var(--foo)' },
+    margin: { all: 'var(--foo)' },
+  },
+  rootC: {
+    // 2️⃣ no custom values
+    padding: ['var(--foo)', 'var(--foo)'],
+  },
+});
+```
+
+## Detailed Design or Proposal
+
+### Ban CSS shorthands in typings
+
+The first part of proposal is to ban CSS shorthands in typings: no shorthand properties = no need to expand them = no issues ✅
+
+Proposed to forbid usage of all CSS shorthands ([list of CSS shorthands exported from MDN](https://csstree.github.io/docs/syntax/#report&noedit&title=CSS%20shorthand%20properties%20by%20MDN%20data&q=JGlzQXJyYXk6ID0%2BICQgIT0gKCQgKyAnJyk7Cm1kbi5wcm9wZXJ0aWVzLmVudHJpZXMoKS4oeyBpZHgsIG5hbWU6IGtleSwgLi4uKHZhbHVlIHwgeyBjb21wdXRlZCwgc3ludGF4IH0pIH0pLltjb21wdXRlZC4kaXNBcnJheSgpXQ%3D%3D&v=ewogICAgdmlldzogJ3RhYmxlJywKICAgIGNvbHM6IHsKICAgICAgICBpZHg6ICd0ZXh0OiMuaW5kZXggKyAxJywKICAgICAgICBjb21wdXRlZDogewogICAgICAgICAgICBoZWFkZXI6ICdwcm9wcycsCiAgICAgICAgICAgIGNvbnRlbnQ6ICd1bDpjb21wdXRlZCcKICAgICAgICB9CiAgICB9LAogICAgbGltaXQ6IGZhbHNlCn0%3D)) except vendor prefixed (i.e. `-ms`, `-webkit`).
+
+```js
+makeStyles({
+  // 🚨 TypeScript will throw
+  rootA: { padding: '4px' },
+  // ✅ TypeScript will pass
+  rootB: { paddingLeft: '4px', paddingRight: '4px', paddingBottom: '4px', paddingTop: '4px' },
+});
+```
+
+### Create macro functions for expansion
+
+As it will be annoying and redundant to use CSS longhands in every case we will have special functions for expansion.
+
+```js
+import { macros } from '@fluentui/react-make-styles';
+
+makeStyles({
+  rootA: { paddingLeft: '4px', paddingRight: '4px', paddingBottom: '4px', paddingTop: '4px' },
+  rootB: {
+    ...macros.padding('4px'),
```

Comment:
For ex-C++ developers, the word macros has a lot of history to it :)  How about calling it shorthand() or fui()?

Reply:
Yeah, it also has overlap with ["Babel macros"](https://github.com/kentcdodds/babel-plugin-macros). I will rename it to `shorthands.*`, that makes more sense to me 👍